File: plugin_platform/plugin/runtime_dispatcher/runtime_registry.py
from .runtime_descriptor import RuntimeDescriptor

class RuntimeRegistry:
    def __init__(self):
        self._descriptors = {}

    def register(self, descriptor: RuntimeDescriptor):
        assert descriptor.runtime_id is not None, "Descriptor runtime_id must not be None"
        self._descriptors[descriptor.runtime_id] = descriptor

    def get(self, runtime_id: str) -> RuntimeDescriptor:
        return self._descriptors.get(runtime_id)

    def get_all(self) -> list:
        # 決定論的ソート (1. priority 降順, 2. runtime_id 昇順)
        return sorted(self._descriptors.values(), key=lambda x: (-x.priority, x.runtime_id))

File: aios/orchestration/registry/RuntimeRegistry.ts
export interface RuntimeDescriptor {
  readonly runtimeName: string;
  readonly version: string;
  readonly capabilities: readonly string[];
}

export class RuntimeRegistry {
  private readonly runtimes = new Map<string, RuntimeDescriptor>();

  /**
   * Registers a runtime capability descriptor.
   */
  public register(descriptor: RuntimeDescriptor): void {
    this.runtimes.set(descriptor.runtimeName, descriptor);
  }

  /**
   * Resolves registered runtime capability descriptors.
   */
  public get(runtimeName: string): RuntimeDescriptor | undefined {
    return this.runtimes.get(runtimeName);
  }

  /**
   * Checks if a target runtime is registered.
   */
  public has(runtimeName: string): boolean {
    return this.runtimes.has(runtimeName);
  }

  /**
   * Clears the current runtime list.
   */
  public clear(): void {
    this.runtimes.clear();
  }
}
